opencti-platform/opencti-graphql/src/manager/telemetryManager.ts

```diff
@@ -0,0 +1,150 @@
+      attributesProcessor: new FilteringAttributesProcessor([]),
+      instrumentSelector: new InstrumentSelector({ type: InstrumentType.HISTOGRAM }),
+      meterSelector: new MeterSelector(),
+    }],
+  }));
+
+  const filigranTelemetryMeterManager = new TelemetryMeterManager(filigranMeterProvider);
+  filigranTelemetryMeterManager.registerFiligranTelemetry();
+  return filigranTelemetryMeterManager;
+};
+
+const fetchTelemetryData = async (filigranTelemetryMeterManager: TelemetryMeterManager) => {
+  try {
+    const context = executionContext('telemetry_manager');
+    // Fetch settings
+    const settings = await getSettings(context) as Settings;
+    // Set filigranTelemetryManager settings telemetry data
+    filigranTelemetryMeterManager.setIsEEActivated(isNotEmptyField(settings.enterprise_edition) ? 1 : 0);
+    filigranTelemetryMeterManager.setEEActivationDate(settings.enterprise_edition);
+    filigranTelemetryMeterManager.setNumberOfInstances(settings.platform_cluster.instances_number);
+    // Get number of active users since fetchTelemetryData() last execution
+    const activUsers = await usersWithActiveSession(SCHEDULE_TIME / 1000 / 60);
+    filigranTelemetryMeterManager.setActivUsersHistogram(activUsers.length);
+  } catch (e) {
+    logApp.error(e, { manager: 'TELEMETRY_MANAGER' });
+  }
+};
+
+const initTelemetryManager = () => {
+  let scheduler: SetIntervalAsyncTimer<[]>;
+  let running = false;
+
+  const telemetryHandler = async (filigranTelemetryMeterManager: TelemetryMeterManager) => {
+    let lock;
+    try {
+      // Lock the manager
+      lock = await lockResource([TELEMETRY_KEY], { retryCount: 0 });
+      running = true;
+      await fetchTelemetryData(filigranTelemetryMeterManager);
+    } catch (e: any) {
+      if (e.name === TYPE_LOCK_ERROR) {
+        logApp.debug('[OPENCTI-MODULE] Telemetry manager already started by another API');
+      } else {
+        logApp.error(e, { manager: 'TELEMETRY_MANAGER' });
+      }
+    } finally {
+      running = false;
+      logApp.debug('[OPENCTI-MODULE] Telemetry manager done');
+      if (lock) await lock.unlock();
+    }
+  };
+
+  return {
+    start: async () => {
+      logApp.info('[OPENCTI-MODULE] Starting telemetry manager');
+      const filigranTelemetryMeterManager = await createFiligranTelemetryMeterManager();
+      // Fetch data periodically
+      scheduler = setIntervalAsync(async () => {
+        await telemetryHandler(filigranTelemetryMeterManager);
+      }, SCHEDULE_TIME);
+    },
+    status: () => {
+      return {
+        id: 'TELEMETRY_MANAGER',
+        enable: booleanConf('telemetry_manager:enabled', false),
+        running,
+      };
+    },
+    shutdown: async () => {
+      logApp.info('[OPENCTI-MODULE] Stopping telemetry manager');
+      if (scheduler) {
+        await clearIntervalAsync(scheduler);
+      }
+      return true;
+    },
+  };
+};
+const telemetryManager = initTelemetryManager();
```

Comment:
Could you use the ManagerModule to write this manager ? same as IndicatorDecayManager

Reply:
done